py-rattler/src/match_spec.rs

```diff
@@ -0,0 +1,104 @@
+use pyo3::{pyclass, pymethods};
+use rattler_conda_types::{MatchSpec, NamelessMatchSpec};
+use std::str::FromStr;
+
+use crate::{error::PyRattlerError, repo_data::package_record::PyPackageRecord};
+
+#[pyclass]
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct PyMatchSpec {
+    inner: MatchSpec,
+}
+
+impl From<MatchSpec> for PyMatchSpec {
+    fn from(value: MatchSpec) -> Self {
+        Self { inner: value }
+    }
+}
+
+impl From<PyMatchSpec> for MatchSpec {
+    fn from(value: PyMatchSpec) -> Self {
+        value.inner
+    }
+}
+
+#[pymethods]
+impl PyMatchSpec {
+    #[new]
+    pub fn __init__(spec: &str) -> pyo3::PyResult<Self> {
+        Ok(MatchSpec::from_str(spec)
+            .map(Into::into)
+            .map_err(PyRattlerError::from)?)
+    }
+
+    /// Returns a string representation of MatchSpec
+    pub fn as_str(&self) -> String {
+        format!("{}", self.inner)
+    }
+
+    /// Match a MatchSpec against a PackageRecord
+    pub fn matches(&self, record: &PyPackageRecord) -> bool {
+        self.inner.matches(&record.clone().into())
```

Comment:
Is this clone really necesary? Could you access the inner through: `record.inner`? Otherwise its perfectly fine to make inner `pub(crate)`.

Reply:
have made the inner `pub(crate)`, it resolves this